src/test/java/com/hillel/items_exchange/controller/UserControllerTest.java

```diff
@@ -352,5 +359,102 @@ void updateUserInfo_shouldUpdateUserDataWithNewChildrenAndPhones() throws Except
                 .andDo(print())
                 .andExpect(jsonPath("$.children", hasSize(1)))
                 .andExpect(jsonPath("$.phones", hasSize(1)));
+            }
+
+    @Test
+    @WithMockUser(username = "trump")
+    @Transactional
+    @DataSet({"database_init.yml", "user/changing_password_or_email/changing_password_or_email_init.yml"})
+    @ExpectedDataSet(value = {"database_init.yml",
+            "user/changing_password_or_email/changing_password_or_email_expected.yml"},
+            ignoreCols = {"password", "email", "lastOnlineTime", "updated"})
+    void updateUserPassword_Successfully() throws Exception {
+        getResultActionsPasswordDto(HttpMethod.PUT, PATH_USER_INFO_CHANGE_PASSWORD,
+                createUserChangePasswordDtoWithCorrectData(), status().isAccepted())
+                .andDo(print());
+        assertTrue(bCryptPasswordEncoder.matches(
+                createUserChangePasswordDtoWithCorrectData().getNewPassword(),
+                "$2a$10$cQvCmoVUukbO/1vTCxeu3OmwJBwAf1S5wabTrL8vUsEZn5DBzlYLW"));
+    }
+
+    @Test
+    @WithMockUser(username = "trump")
+    @Transactional
+    @DataSet({"database_init.yml", "user/changing_password_or_email/changing_password_or_email_init.yml"})
+    void updateUserPassword_WhenOldPasswordWrong_ShouldThrowInvalidDtoException() throws Exception {
+        final MvcResult mvcResult = getResultActionsPasswordDto(HttpMethod.PUT,
+                PATH_USER_INFO_CHANGE_PASSWORD,
+                createUserChangePasswordDtoWithWrongOldPassword(), status().isBadRequest())
+                .andDo(print())
+                .andReturn();
+        assertTrue(mvcResult.getResponse().getContentAsString()
+                .contains(getExceptionMessageSource("not.your.password")));
+    }
+
+    @Test
+    @WithMockUser(username = "trump")
+    @Transactional
+    @DataSet({"database_init.yml", "user/changing_password_or_email/changing_password_or_email_init.yml"})
+    void updateUserPassword_WhenPasswordConfirmationWrong_ShouldThrowIllegalArgumentException()
+            throws Exception {
+        final MvcResult mvcResult = getResultActionsPasswordDto(HttpMethod.PUT,
+                PATH_USER_INFO_CHANGE_PASSWORD,
+                createUserChangePasswordDtoWithWrongPasswordConfirmation(), status().isBadRequest())
+                .andDo(print())
+                .andReturn();
+
+        final String message = Objects.requireNonNull(mvcResult.getResolvedException()).getMessage();
+
+        assertTrue(message.contains(getExceptionMessageSource("invalid.confirm.password")));
+    }
+
+    @Test
+    @WithMockUser(username = "trump")
+    @Transactional
+    @DataSet({"database_init.yml", "user/changing_password_or_email/changing_password_or_email_init.yml"})
+    @ExpectedDataSet(value = {"database_init.yml",
+            "user/changing_password_or_email/changing_password_or_email_expected.yml"},
+            ignoreCols = {"password", "lastOnlineTime", "updated"})
```

Comment:
password is encoded. It cannot be compared because every time we get different data. It can be compared only with bCryptPasswordEncoder.matches(String encodedPassword, String password)